### PR TITLE
Document 'person/{login}/token' post method

### PR DESCRIPTION
**Pull request by saraycp**
Using OpenAPI.

Co-authored-by: Eduardo Navarro <enavarro@suse.com>

-----------------

If this PR requires any particular action or consideration before deployment,
please check the reasons or add your own to the list:

* [ ] Requires a database migration that can cause downtime. Postpone deployment until maintenance window[1].
* [ ] Contains a data migration that can cause temporary inconsistency, so should be run at a specific point of time.
* [ ] Changes some configuration files (e.g. options.yml), so the changes have to be applied manually in the reference server.
* [ ] A new Feature Toggle[2] should be enabled in the reference server.
* [ ] Proper documentation or announcement has to be published upfront since the introduced changes can confuse the users.

[1] https://github.com/openSUSE/open-build-service/wiki/Deployment-of-build.opensuse.org#when-there-are-migrations
[2] https://github.com/openSUSE/open-build-service/wiki/Feature-Toggles-%28Flipper%29#you-want-real-people-to-test-your-feature

### CODE REVIEW

File: src/api/public/apidocs-new/paths/person_login_token.yaml [hunk 1]
```diff
@@ -40,6 +40,87 @@ get:
...
+  tags:
+    - Person
+
+post:
+  summary: Create a new authentication token for a person.
+  description: |
+    Create a new authentication token for a person passed as parameter.
+
+    The token may be limited for a specific package. In this case the query parameters
+    'project' and 'package' should be provided.
```
Comment (krauselukas) [inline src/api/public/apidocs-new/paths/person_login_token.yaml:52]:
```suggestion
    `project` and `package` should be provided.
```

Comment (krauselukas) [reply src/api/public/apidocs-new/paths/person_login_token.yaml:52]: I would use the code style for parameters in the description, it highlights it a bit

File: src/api/public/apidocs-new/paths/person_login_token.yaml [hunk 2]
```diff
@@ -40,6 +40,87 @@ get:
...
+        Package name. Should be provided, together with the project query parameter,
+        to limit the token for a specific package.
+      example: gchz
+    - in: query
+      name: operation
+      schema:
+        type: string
+        enum: [runservice, rebuild, release]
+      description: |
+        Operation indicates the kind of token is going to be created.
```
Comment (krauselukas) [inline src/api/public/apidocs-new/paths/person_login_token.yaml:81]:
```suggestion
        Operation indicates the kind of token that is going to be created.
```

File: src/api/public/apidocs-new/paths/person_login_token.yaml [hunk 3]
```diff
@@ -40,6 +40,87 @@ get:
...
+    - basic_authentication: []
+  parameters:
+    - $ref: '../components/parameters/login.yaml'
+    - in: query
+      name: project
+      schema:
+        type: string
+      description:  |
+        Project name. Should be provided, together with the package query parameter,
+        to limit the token for a specific package.
```
Comment (krauselukas) [inline src/api/public/apidocs-new/paths/person_login_token.yaml:65]:
```suggestion
        to limit the token to a specific package.
```

File: src/api/public/apidocs-new/paths/person_login_token.yaml [hunk 4]
```diff
@@ -40,6 +40,87 @@ get:
...
+      example: home:hans
+    - in: query
+      name: package
+      schema:
+        type: string
+      description: |
+        Package name. Should be provided, together with the project query parameter,
+        to limit the token for a specific package.
```
Comment (krauselukas) [inline src/api/public/apidocs-new/paths/person_login_token.yaml:73]:
```suggestion
        to limit the token to a specific package.
```

File: src/api/public/apidocs-new/paths/person_login_token.yaml [hunk 5]
```diff
@@ -40,6 +40,87 @@ get:
           example:
             code: view_user_not_authorized
             summary: 'You are not authorized to view this User.'
+  tags:
+    - Person
+
+post:
+  summary: Create a new authentication token for a person.
+  description: |
+    Create a new authentication token for a person passed as parameter.
```
Comment (krauselukas) [inline src/api/public/apidocs-new/paths/person_login_token.yaml:49]:
```suggestion
    Create a new authentication token for a person.
```

Comment (krauselukas) [reply src/api/public/apidocs-new/paths/person_login_token.yaml:49]: the passed as parameter part sound a bit strange to me, I think its not needed :)

Comment (saraycp) [reply src/api/public/apidocs-new/paths/person_login_token.yaml]: Yes. Better without that part.

I addressed this and the rest of your suggestions. Please have another look @krauselukas

**Review by krauselukas — APPROVED** [review]

**Review by danidoni — APPROVED** [review]

**Pull request merged**